Code/Framework/AzCore/AzCore/RTTI/BehaviorContext.h

```diff
@@ -206,25 +216,48 @@ namespace AZ
     public:
         AZ_CLASS_ALLOCATOR(BehaviorDefaultValue, AZ::SystemAllocator, 0);
 
+        const BehaviorValueParameter& GetValue() const
+        {
+            return m_value;
+        }
+
+        BehaviorValueParameter m_value;
+    };
+
+    template<typename Value>
+    class BehaviorDefaultValueTyped
+        : public BehaviorDefaultValue
+    {
+    public:
         /**
         * Create a default value for a specific method parameter. The Default values is stored by value
         * in a temp storage, so currently there is limit the \ref BehaviorValueParameter temp storage, we
         * can easily change that if it became a problem.
         */
-        template<typename Value>
-        BehaviorDefaultValue(Value&& value)
+        BehaviorDefaultValueTyped(Value&& value)
```

Comment:
This is an rvalue only constructor.
It is a forwarding constructor as the `Value` type is not a template type in regard to this function